src/platform/plugins/shared/data/common/search/search_source/search_source.ts

```diff
@@ -1223,23 +1223,16 @@ export class SearchSource {
   }
 
   parseActiveIndexPatternFromQueryString(queryString: string): string[] {
-    let m;
     const indexPatternSet: Set<string> = new Set();
-    const regex = /\s?(_index)\s?:\s?[\'\"]?(\w+\-?\*?)[\'\"]?\s?(\w+)?/g;
+    //  Regex to capture full index names including dashes, numbers, and periods
+    const indexNameRegExp = /\s?(_index)\s*:\s*['"]?([^\s'"]+)['"]?/g;
```

Comment:
Not blocking but just food for thought.

This regex will not care about matching `"` and `'` which means it will match something like: 

```
_index: "logs-2024-06-27'
```

This is probably not needed as it was like this previously.

But you could fix it to match the pair such that the first `"` is paired with a `"`, like so...

```suggestion
    const indexNameRegExp = /\s?(_index)\s*:\s*(['"]?)([^\s'"]+)\2/g;
```

If you care to make this change you would need to also update the match code to `const indexName = match[3]`.

Reply:
Thx @nickofthyme!  Since we're on it , let do it, let's make it better
https://github.com/elastic/kibana/pull/221084/commits/4aa6563339e65c2c7083aff95531cebd1b2a4459